homeassistant/components/template/lock.py

```diff
@@ -129,24 +135,58 @@ def _update_state(self, result):
 
         self._state = None
 
+    @property
+    def code_format(self) -> str | None:
+        """Regex for code format or None if no code is required."""
+        if self._code_format is not None:
+            return self._code_format
+
+        return super().code_format
```

Comment:
We don't need to call super here
```suggestion
        return self._code_format
```